doc/whatsnew/fragments/1727.false_negative.1

```diff
@@ -0,0 +1,19 @@
+``used-before-assignment`` is now emitted when relying on variable assignments
+that were not exhaustively made in every if/else branch.
+
+If you rely on a pattern like this:
+```
+if guarded():
+    var = 1
+
+if guarded():
+    print(var)  # now emits used-before-assignment
+```
```

Comment:
@milosivanovic This is what I'm worried people will complain about. Do you have a view?

Reply:
I think the warning is reasonable. It's a good reminder for the developer to ensure 'var' has a default value at a higher level, which would be good practice. That should stop the warning from happening, right?